application/src/test/java/org/thingsboard/server/controller/BaseEntityViewControllerTest.java

```diff
@@ -308,17 +315,66 @@ public void testGetTenantEntityViewsByName() throws Exception {
                 },
                 new TextPageLink(4, name2));
         Assert.assertFalse(pageData.hasNext());
-        Assert.assertEquals(0, pageData.getData().size());
+        assertEquals(0, pageData.getData().size());
+    }
+
+    @Test
+    public void testTheCopyOfAttrsThatMatchWithDeviceCriteriaForTheView() throws Exception {
+
+        String viewDeviceId = testDevice.getId().getId().toString();
+        DeviceCredentials deviceCredentials
+                = doGet("/api/device/" + viewDeviceId + "/credentials", DeviceCredentials.class);
+
+        assertEquals(testDevice.getId(), deviceCredentials.getDeviceId());
+
+        String accessToken = deviceCredentials.getCredentialsId();
+        assertNotNull(accessToken);
+
+        String clientId = MqttAsyncClient.generateClientId();
+        MqttAsyncClient client = new MqttAsyncClient("tcp://localhost:1883", clientId);
+
+        MqttConnectOptions options = new MqttConnectOptions();
+        options.setUserName(accessToken);
+        client.connect(options);
+        Thread.sleep(3000);
+
+        MqttMessage message = new MqttMessage();
+        message.setPayload(("{\"caValue1\":\"value1\", \"caValue2\":true, \"caValue3\":42.0, \"caValue4\":73}").getBytes());
+        client.publish("v1/devices/me/attributes", message);
+        Thread.sleep(1000);
+
+        List<String> actualTelemetry =
```

Comment:
actualAttributes instead of actualTelemetry would improve the "readability" of the test